Update payment docs (#11751)
woocommerce/woocommerce-gutenberg-products-block: docs/internal-developers/block-client-apis/checkout/checkout-flow-and-events.md

#### Success

> A successs response should be given when there are the user entered data is correct and the payment checks are successful. A response is considered a success response when it at a minimum is an object with this shape:

Reviewer: This isn't a cohesive sentence.

```suggestion
A successful response should be given when the user's entered data is correct and the payment checks are successful. A response is considered successful if, at a minimum, it is an object with this shape:
```
